grim.go

```diff
@@ -189,19 +191,31 @@ func onHook(configRoot string, config *effectiveConfig, hook hookEvent, action h
 	writeHookEvent(resultPath, hook)
 
 	// TODO: do something with the err
-	notify(config, hook, "", GrimPending)
+	message, _ := notify(config, hook, "", GrimPending)
+	if logger != nil {
+		logger.Print("grim pending: " + message + "\n")
+	}
 
 	result, ws, err := action(configRoot, resultPath, config, hook)
 	if err != nil {
-		notify(config, hook, ws, GrimError)
+		message, _ = notify(config, hook, ws, GrimError)
```

Comment:
in the case of this function you are passing in a logger to do the logging, in the case of hte notify function you are returning a message, which the logger is then logging here.

Why did you choose to do it that way?  Another way would be to pass the logger all the way to the notify function for instance.